### DISCUSSION TITLE
[help wanted] seems httpgd plot backend will freeze vscode-r extension

### DISCUSSION THREAD

**Discussion by TTTPOB** [REditorSupport/vscode-R#950] [Q&A] [answered]
I dont know how to demonstrate this issue
If I do a plot with many (well, not so many in fact) elements on it, say a heatmap of 30*200, the vscode-R extension will freeze for about 30s. I mean hit ctrl + enter will not send code to terminal, and httpgd plot window shows the plot very slowly. after the plot is fully shown, I can send code to the terminal again.
I dont know what information that I can provide to identify where the issue lies, so I think I need some help and thus not putting it in the issue section.

**Comment by renkun-ken** [comment]
Yes, your observation is definitely valid and I experience such problem too.

I guess it must be caused by that too many svg elements in a WebView will cause VS Code to render slowly and user interaction with UI and terminal will become sluggish.

@nx10, do you think if there is anything we could do about it?

**Comment by nx10** [comment, marked as answer]
This seems to be a client issue. When I run this code in the vscode-r viewer the UI becommes laggy:

```R
mat <- matrix(rnorm(30*200), nrow=30)
heatmap(mat)
```

However when it is closed and I open the httpgd client with e.g. `httpgd::hgd_view()` it runs smoothly. 

The vscode-r viewer actually handles plots a bit differently and will load and append the SVG element directly to the editor DOM which might be the cause for the performance issues. @ManuelHentschel what do you think would be the best way to fix this?

**Reply by ManuelHentschel** [reply]
Yup, seems to be the client. Re-adding a delay that we accidently removed in https://github.com/REditorSupport/vscode-R/pull/823 seems to fix this for me: https://github.com/REditorSupport/vscode-R/pull/956

**Reply by nx10** [reply]
Good job. I will see if I can implement the delay server side when I finally get around to rewrite the threading logic.

**Reply by TTTPOB** [reply]
great! it works again, I can finally use the httpgd backend
